### PR TITLE
Bug #3320 - Restructure repo structure

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

File: make.js [hunk 1]
```diff
@@ -296,20 +258,16 @@ function checkHTML() {
...
 
-  find([
-    PUBLIC_DIR,
-    join( SRC_DIR, "dialog", "dialogs" ),
-    join( SRC_DIR, "layouts" ),
-    join( SRC_DIR, "editor" ),
-    join( SRC_DIR, "ui", "webmakernav" ),
-    TEMPLATES_DIR ] ).filter( function( file ) {
-    return file.match( /\.html$/ );
+  find( 'public/' ).filter( function( file ) {
```
Comment (mjschranz) [inline make.js]: Space nit and double quotes.

File: make.js [hunk 2]
```diff
@@ -145,17 +118,17 @@ function checkCSS() {
   echo( "" );
   echo( "# Linting CSS files" );
 
-  find( SLICE.call( arguments ) ).filter(function( filename ) {
-    return (/\.css$/).test( filename );
+  find( 'public/' ).filter(function( filename ) {
```
Comment (mjschranz) [inline make.js]: We seem to have a nice mix of double/single quotes. Stay double.

File: make.js [hunk 3]
```diff
@@ -145,17 +118,17 @@ function checkCSS() {
...
     checkCSSFile( filename, warnings, errors );
   });
 
 }
 
-function checkJS(){
-  // Takes a string or an array of strings referring to directories.
-  var dirs = SLICE.call( arguments );
+function checkJS() {
+  var dirs = [ '*.js', 'config/', 'lib/', 'public/src/', 'public/templates/', 'routes/', 'test/' ];
```
Comment (mjschranz) [inline make.js]: Double quotes.

File: make.js [hunk 4]
```diff
@@ -332,66 +290,29 @@ function buildCSS(compress) {
...
+    "public/css/butter.ui",
+    "public/css/transitions",
+    "public/css/embed",
+    "public/css/embed-shell",
+    "public/src/ui/webmakernav/webmakernav",
+    "public/templates/assets/css/jquery-ui/jquery.ui.butter",
+    "public/templates/assets/plugins/twitter/popcorn.twitter",
+    "public/templates/assets/plugins/wikipedia/popcorn.wikipedia",
+    "public/templates/basic/style"
+  ].forEach(function(file) {
```
Comment (mjschranz) [inline make.js]: space nit.

File: make.js [hunk 5]
```diff
@@ -411,18 +332,24 @@ function buildJS( version, compress ){
...
+  echo( "" );
+  echo( "# Optimizing JS Files" );
+
+  echo( "## public/src/butter.js" );
   result = exec(RJS + ' -o tools/build-butter.js ' + doCompress, {silent: true});
   if (!!result.code) {
     echo(result.output);
   }
-  stampVersion( version, 'dist/src/butter.js' );
+  stampVersion( version, 'dist/public/src/butter.js' );
```
Comment (mjschranz) [inline make.js]: Double Quotes.

File: make.js [hunk 6]
```diff
@@ -432,27 +359,7 @@ function buildJS( version, compress ){
...
-  });
-
-  server.stderr.on( 'data', function( data ) {
-    process.stderr.write( "" + data );
-  });
-
-  server.on( 'exit', function( code ) {
-    console.log( 'server process exited with code ' + code );
-  });
+  require("./server.js");
```
Comment (mjschranz) [inline make.js]: Space.

File: make.js [hunk 7]
```diff
@@ -463,35 +370,45 @@ target.deploy = function(){
   var compress = env.UNMINIFIED !== "1",
       version = env.VERSION;
 
-  clean();
+  rm('-fr', DIST_DIR);
```
Comment (mjschranz) [inline make.js]: Space, double quotes.

File: make.js [hunk 8]
```diff
@@ -463,35 +370,45 @@ target.deploy = function(){
...
+  mkdir('-p', DIST_DIR);
```
Comment (mjschranz) [inline make.js]: ^

File: make.js [hunk 9]
```diff
@@ -1,63 +1,36 @@
...
-    date: (new Date()).toJSON(),
-    version: env.VERSION || 'development',
-    popcorn: gitDescribe( 'external/popcorn-js/' ),
-    butter: gitDescribe( '.' )
-  }, null, 2 ).to( versionConfig );
-}
-
 // To supress CSS warnings/errors for a particular line, end the line
-// with a comment indicating you want CSS Lint to ignore this line's
+// with a comment indicating you want CSS Lint to ignore this line"s
```
Comment (cadecairos) [inline make.js]: the `'` shouldn't have been replaced on this line

File: make.js [hunk 10]
```diff
@@ -84,16 +57,16 @@ function checkCSSFile( filename, warnings, errors ) {
...
+    " --errors=" + errors +
+    " --quiet --format=compact" +
+    " " + filename, { silent: true } ).output.split( /\r?\n/ )
     .forEach( function( line ) {
       if( !line ) {
         return;
       }
 
-      // Some warnings don't refer to a line, e.g.
+      // Some warnings don"t refer to a line, e.g.
```
Comment (cadecairos) [inline make.js]: same here

File: make.js [hunk 11]
```diff
@@ -1,61 +1,34 @@
...
-    join = path.join,
+    normalize = function( p ){ return "'" + path.normalize( p ) + "'"; },
     // Make Windows happy, use `node <path>`
     nodeExec = function( p ){ return 'node "' + p + '"'; },
     pythonExec = function( p ){ return 'python "' + p + '"'; },
-    SLICE = Array.prototype.slice,
 
     JSLINT = nodeExec( normalize( "./node_modules/jshint/bin/hint" ) ),
-    HTML5LINT = pythonExec( normalize( "./external/html5-lint/html5check.py" ) ),
+    HTML5LINT = pythonExec( normalize( "./public/external/html5-lint/html5check.py" ) ),
```
Comment (cadecairos) [inline make.js:10]: html5check.py is failing when executed. it needs to be updated

Comment (jbuck) [inline make.js:10]: Update your git submodules: `git submodule update --init`

Comment (cadecairos) [inline make.js:10]: nope, chuck testa

File: make.js [hunk 12]
```diff
@@ -411,101 +332,98 @@ function buildJS( version, compress ){
...
+    "config",
+    "lib",
+    "npm-shrinkwrap.json",
+    "package.json",
+    "routes",
+    "server.js",
+    "views"
+  ], "dist/");
+
+  // Selectively copy things from public/
```
Comment (jbuck) [inline make.js:391]: Document why we selectively copy things

**Pull request merged**